### DISCUSSION TITLE
Add light/dark theme to application

### DISCUSSION THREAD

**Discussion by noahfraiture** [prasanthrangan/hyprdots#1714] [Q&A] [answered]
Hello, how can I add a light and dark theme for an app ? I use helix as my editor and I would to switch from multiple theme based on the selected theme or at least between light/dark to match the terminal color. 
I already did it once but it was with the old script and configuration system

**Comment by kRHYME7** [comment]
Checkout this image

![image](https://github.com/user-attachments/assets/d982346a-08c3-413d-8495-0a9dcca44986)

For each hypr.theme file in the theme directory ( ` /home/khing/.config/hyde/themes/` ), we got the line ``` exec = gsettings set org.gnome.desktop.interface color-scheme 'prefer-dark' ```
 
 This dictates mostly of what color-scheme the themes will use. all gtk and I guess qt apps respects this. 
 Therefore your app should follow this. If not check you apps settings instead 




If you remembered old implementation the the theme.conf is now being templated as hypr.theme.

**Reply by noahfraiture** [reply]
Oh thank you, and where are these file executed ? For instance if I want to add a 'helix.theme' for better organisation

**Reply by kRHYME7** [reply]
I do not know helix but how do you change themes on it? Do we have commands to change the theme?

**Reply by noahfraiture** [reply]
I simply need a bash command. I change a symlink then I think I can reload the theme config with a signal. But I would like to not put that in the hypr.theme file if possible since it isn't really related to hyprland directly

**Reply by kRHYME7** [reply, marked as answer]
You got 3 ways to do it

### 1. Either add the command as exec in `hypr.theme`

` exec = your command `

This is fine because themes are installed from a separate repo. So if you update the dot files it will not touch the theme. You can update the theme only if you use ` themepatcher.sh ` or ` Hyde theme patch `


### 2.a Creating a  Template `helix.theme`

Basically you can add a helix.theme. 
* Inside that file should  have a "header" at line 1 be structured like this:

` $HOME/path/to/tgt/file | cmd to execute after populating `

* Below it should be the target file's contents. So this way you don't need to do symlinks. 
* You can just add the contents in that template and 'wallbash.sh'  will populate that content to the target file during the theme/wall changes.

Note that to make this functional you need to have a fallback template inside ` ~/.config/hyde/wallbash/Wall-Dcol/` so you  add a default template here name it `helix.dcol`  (same name different suffix) *ps you can use the colors from .dcol files see other .dcol files as inspiration.


### 2.b Creating a dummy Template `helix.theme` to execute a command

This approach is workaround ish. If you want to stick to your method of sym links;
You will make a `helix.theme` but to only execute a command.

* Create a `helix.theme` With this header 

` /tmp/dump | your command  `

* tmp/dump will be the dump file for the outputs which will be blank
* `your command` will be executed after writing in `/tmp/dump/`


Note that you can name it whatever you want with the requirements of the target files in the header should have an existing directory name.

**Reply by noahfraiture** [reply]
Great thank you !
